test/e2e/framework/framework.go

```diff
@@ -142,6 +147,8 @@ func (f *Framework) BeforeEach() {
 	assert.Nil(ginkgo.GinkgoT(), err, "updating ingress controller pod information")
 
 	f.WaitForNginxListening(80)
+
+	f.HTTPBunIP = f.NewHttpbunDeployment()
```

Comment:
What if we always deploy httpbun as part of the framework bootstrap? Instead of deploying on each test?

The way you are doing here you are deploying an httpbun always, I would say maybe when creating the test, deploy a single httpbun and we use in all the tests, as we don't mutate anything between tests

Reply:
How would this work for routing as each test gets run in its own namespace in parallel, So when creating the ingress we would always have to do something like create an external service to the httpbun as it wouldn't be running in the tests namespace

Reply:
I went for the idea disabling it by default but having the ability to enable it, deploying it with the bootstrapping seemed a bit complex as all tests are generated with their own namespace and it would be a bit complex to start setting up routing between namespaces and might start not being true tests